tensorflow/contrib/ignite/python/ops/ignite_dataset_ops.py

```diff
@@ -0,0 +1,763 @@
+    binary_type_name = self.__parse_string()
+    self.__parse_string()   # Affinity field name
+
+    fields = []
+    for _ in range(self.read_int()):
+      field_name = self.__parse_string()
+      field_type_id = self.read_int()
+      field_id = self.read_int()
+
+      field = BinaryField(field_name, field_type_id, field_id)
+      fields.append(field)
+
+    is_enum = self.read_byte()
+    if is_enum == 1:
+      raise Exception("Enum fields are not supported yet")
+
+    schema_cnt = self.read_int()
+    for _ in range(schema_cnt):
+      self.read_int()       # Schema id
+      field_cnt = self.read_int()
+      self.skip(field_cnt * 4)
+
+    return BinaryType(binary_type_id, binary_type_name, fields)
+
+  def __parse_string(self):
+    """Parses string."""
+    header = self.read_byte()
+    if header == 9:
+      length = self.read_int()
+      return self.read_data(length).decode("utf-8")
+    if header == 101:
+      return None
+    raise Exception("Unknown binary type when expected string [type_id=%d]" \
+        % header)
+
+class IgniteDataset(Dataset):
+  """Apache Ignite is a memory-centric distributed database, caching, and
+     processing platform for transactional, analytical, and streaming workloads,
+     delivering in-memory speeds at petabyte scale. This contrib package
+     contains an integration between Apache Ignite and TensorFlow. The
+     integration is based on tf.data from TensorFlow side and Binary Client
+     Protocol from Apache Ignite side. It allows to use Apache Ignite as a
+     datasource for neural network training, inference and all other
+     computations supported by TensorFlow. Ignite Dataset is based on Apache
+     Ignite Binary Client Protocol.
+  """
+
+  def __init__(self, cache_name, host="localhost", port=10800, local=False,\
+    part=-1, page_size=100, username=None, password=None, certfile=None,\
+    keyfile=None, cert_password=None):
+    """Create a IgniteDataset.
+
+    Args:
+      cache_name: Cache name to be used as datasource.
+      host: Apache Ignite Thin Client host to be connected.
+      port: Apache Ignite Thin Client port to be connected.
+      local: Local flag that defines to query only local data.
+      part: Number of partitions to be queried.
+      page_size: Apache Ignite Thin Client page size.
+      username: Apache Ignite Thin Client authentication username.
+      password: Apache Ignite Thin Client authentication password.
+      certfile: File in PEM format containing the certificate as well as
+        any number of CA certificates needed to establish the certificate’s
+        authenticity.
+      keyfile: File containing the private key (otherwise the private key
+        will be taken from certfile as well).
+      cert_password: Password to be used if the private key is encrypted and a
+        password is necessary.
+    """
+    super(IgniteDataset, self).__init__()
+
+    with IgniteClient(host, port, username, password, certfile, keyfile,\
```

Comment:
There is no need for a line continuation character here (or in the cases below) because it is implied by the open parentheses.

Reply:
Thanks, fixed.